### ISSUE TITLE
Pallas build duplicates work

### ISSUE THREAD

**Issue by olupton** [NVIDIA/JAX-Toolbox#467]
When building wheels for Pallas the CI jobs are duplicating work.
e.g. in https://github.com/NVIDIA/JAX-Toolbox/actions/runs/7475866231/job/20344957851#step:10:428 then parts of the Dockerfile are being re-executed that should be cached from the "Build mealkit image" part of the same job.
For example, in the PAX equivalent this caching works: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/7475866243/job/20344960164#step:10:181
I have not been able to reproduce this locally, where the run with `--target final` does hit the cache.

**Comment by yhtang** [comment]
When you reproduce that locally, did you use the same buildkit version as in the GitHub Action?

**Comment by olupton** [comment]
On the machine I tried locally I had
```
$ docker --version
Docker version 24.0.7, build afdd53b
$ docker buildx version
github.com/docker/buildx v0.11.2 9872040
```
whereas it looks like the CI has 24.0.0 and v0.12.0 542e5d810e4a1a155684f5f3c5bd7e797632a12f.
So no, they're close but don't match.

**Comment by nouiz** [comment]
Note, pallas is included in the JAX container.

**Comment by olupton** [comment]
Yes. There is still a separate `triton` container containing JAX-Triton, which is arguably where we would expect this bug to show up if it were still valid. However, the caching seems to be working fine in the `triton` container build. I should have cited https://github.com/NVIDIA/JAX-Toolbox/pull/798 for why.
